Template documentation emphasises container usage (#539)
nf-core/tools: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/docs/usage.md

Several generic profiles are bundled with the pipeline which instruct the pipeline to use software packaged using different methods (Docker, Singularity, Conda) - see below.

> > We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when these are not possible, conda is well supported.

Reviewer: ```suggestion
> We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when these are not possible, Conda is also supported.
```

> > We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when these are not possible, Conda is also supported.

Reviewer: ```suggestion
> We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when this is not possible, Conda is also supported.
```

Note that multiple profiles can be loaded, for example: `-profile test,docker` - the order of arguments is important!
They are loaded in sequence, so later profiles can overwrite earlier profiles.

If `-profile` is not specified, the pipeline will run locally and expect all software to be installed and available on the `PATH`. This is _not_ recommended.

* `docker`
  * A generic configuration profile to be used with [Docker](http://docker.com/)
  * Pulls software from dockerhub: [`{{ cookiecutter.name_docker }}`](http://hub.docker.com/r/{{ cookiecutter.name_docker }}/)
* `singularity`
  * A generic configuration profile to be used with [Singularity](http://singularity.lbl.gov/)
  * Pulls software from DockerHub: [`{{ cookiecutter.name_docker }}`](http://hub.docker.com/r/{{ cookiecutter.name_docker }}/)
* `conda`

>   * Recommended only if Docker and Singularity not possible

Reviewer: ```suggestion
  * Recommended only if it's not possible to run the pipeline wirh Docker or Singularity.
```

>   * Please only use Conda as a last resort i.e. it's not possible to run the pipeline with Docker or Singularity.

Reviewer: ```suggestion
  * Please only use Conda as a last resort i.e. when it's not possible to run the pipeline with Docker or Singularity.
```

>   * A generic configuration profile to be used with [conda](https://conda.io/docs/)

Reviewer: ```suggestion
  * A generic configuration profile to be used with [Conda](https://conda.io/docs/)
```
